sdks/python/apache_beam/testing/test_stream_test.py

```diff
@@ -528,6 +529,56 @@ def process(
 
     p.run()
 
+  def test_roundtrip_proto(self):
+    test_stream = (TestStream()
+                   .advance_processing_time(1)
+                   .advance_watermark_to(2)
+                   .add_elements([1, 2, 3])) # yapf: disable
+
+    p = TestPipeline(options=StandardOptions(streaming=True))
+    p | test_stream
+
+    pipeline_proto, context = p.to_runner_api(return_context=True)
+
+    for t in pipeline_proto.components.transforms.values():
+      if t.spec.urn == common_urns.primitives.TEST_STREAM.urn:
+        test_stream_proto = t
+
+    self.assertTrue(test_stream_proto)
+    roundtrip_test_stream = TestStream().from_runner_api(
+        test_stream_proto, context)
+
+    self.assertListEqual(test_stream._events, roundtrip_test_stream._events)
+    self.assertSetEqual(
+        test_stream.output_tags, roundtrip_test_stream.output_tags)
+    self.assertEqual(test_stream.coder, roundtrip_test_stream.coder)
+
+  def test_roundtrip_proto_multi(self):
+    test_stream = (TestStream(output_tags=['a', 'b'])
```

Comment:
I don't think you should need to specify output_tags here since they will get added via the add/advance calls below.

Reply:
Gotcha, removed the output_tags